magento/magento2
# Added getScopeCacheKey in order to define the right cache key based on store ID

### Description (*)
This PR adds an small improvement to the instance cache into the categories repository cause it allows to receive store id or store code for attributes data retrieval but at the same time this store id / code is used as cache key which make it a double unnecessary category load when you request the category by same given store, first providing the store id and then providing store code.

### Manual testing scenarios (*)
1. Into your own class load the category repository via di.
2. Use the category repository to get a given category providing an specific store ID. Check the given category got stored into the cached instances array of the repository under the store ID.
3. Use the category repository to get a given category (same category as previous point) providing a specific store Code (same store as previous point). Error should be thrown with this fix, otherwise it was loading again the category filtering by store code which is incorrect based on PHP Docs into the interface.

### Questions or comments
Have in mind this issue is not reproducible from REST API given it enforces to use the store ID based on the PHP Docs from the interface.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#28142: Added getScopeCacheKey in order to define the right cache key based on store ID
